### DISCUSSION TITLE
How to authenticate via websockets

### DISCUSSION THREAD

**Discussion by Q16solver** [Hebilicious/authjs-nuxt#110] [Q&A]
I have a feeling it’s more of a nuxt issue, but getServerSession is only via http and I’m not sure if there’s any workarounds/examples for using the ws package

**Comment by Hebilicious** [comment]
@Q16solver I have never tried. I'm not sure if this a Nuxt/Auth.js issue, can you write some example of WS authentication with Nuxt but without auth.js ?

**Reply by Q16solver** [reply]
@Hebilicious  I don't know about nuxt, but it does seem like next.js supports auth with ws on trpc with next-auth https://github.com/trpc/examples-next-prisma-websockets-starter/blob/main/src/server/context.ts
the issue might be the incompatibility with the h3 event object that nuxt uses which cannot be constructed on websockets?
Which is what I currently have https://github.com/Esposter/Esposter/blob/main/server/trpc/context.ts

**Reply by Hebilicious** [reply]
H3 currently doesn't support websocket indeed. That being said, it's being worked on as we speak.
